### DISCUSSION TITLE
Set content type metadata during uploading directory to S3

### DISCUSSION THREAD

**Discussion by asemeria** [aws/aws-sdk-java-v2#4190] [Migration from v1 to v2]
I need to set the metadata for some files according to a naming logic.
I would like to accomplish that with  S3TransferManager (https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/transfer-manager.html), but I cannot find a method to do it.

With the TransferManager library, this was achievable through the method:

`MultipleFileUpload uploadDirectory(String bucketName, String virtualDirectoryKeyPrefix, File directory, boolean includeSubdirectories, ObjectMetadataProvider metadataProvider)`

Using the metadataProvider, it was possible to set the metadata for each file.

How can I achieve the same functionality with S3TransferManager?
Thank you

**Comment by debora-ito** [comment]
You can use `uploadFileRequestTransformer()` to modify each PutObjectRequest that is created for the upload -

https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/transfer/s3/model/UploadDirectoryRequest.Builder.html#uploadFileRequestTransformer(java.util.function.Consumer)

Would this be sufficient, or would you like to see something more close to TransferManager v1 like directly providing a `ObjectMetadataProvider`?
